packages/babel-plugin-proposal-decorators/src/transformer-legacy.js

```diff
@@ -287,4 +287,19 @@ export default {
       ]),
     );
   },
+
+  CallExpression(path, state) {
+    // insert assignment if class properties are in strict mode
+    if (path.node.callee.name !== "_defineProperty") return;
```

Comment:
This is not safe. If a user already has a `_defineProperty` binding in their code, Babel will use another name.

The check should look more like:

```js
if (path.node.arguments.length !== 2) return;
if (!WARNING_CALLS.has(path.node.arguments[2])) return;

// If the class properties plugin isn't enabled, this line will add an unused helper
// to the code. It's not ideal, but it's ok since the configuration is not valid anyway.
if (path.node.callee.name !== state.addHelper("defineProperty").name) return;
```